WordPress/Classes/Services/SharingService.swift

```diff
@@ -106,50 +130,72 @@ open class SharingService: LocalCoreDataService {
     ///     - success: An optional success block accepting no parameters.
     ///     - failure: An optional failure block accepting an NSError parameter.
     ///
-    @objc open func updateSharedForBlog(_ blog: Blog,
+    @objc func updateSharedForBlog(
+        _ blog: Blog,
         shared: Bool,
         forPublicizeConnection pubConn: PublicizeConnection,
         success: (() -> Void)?,
-        failure: ((NSError?) -> Void)?) {
+        failure: ((NSError?) -> Void)?
+    ) {
+        typealias PubConnUpdateResult = (oldValue: Bool, siteID: NSNumber, connectionID: NSNumber, service: String, remote: SharingServiceRemote?)
 
-            if pubConn.shared == shared {
-                success?()
-                return
-            }
-
-            let oldValue = pubConn.shared
-            pubConn.shared = shared
-            ContextManager.sharedInstance().save(managedObjectContext)
+        let blogObjectID = blog.objectID
+        coreDataStack.performAndSave({ context -> PubConnUpdateResult in
+            let blogInContext = try context.existingObject(with: blogObjectID) as! Blog
+            let pubConnInContext = try context.existingObject(with: pubConn.objectID) as! PublicizeConnection
+            let oldValue = pubConnInContext.shared
+            pubConnInContext.shared = shared
+            return (
+                oldValue: oldValue,
+                siteID: pubConnInContext.siteID,
+                connectionID: pubConnInContext.connectionID,
+                service: pubConnInContext.service,
+                remote: self.remoteForBlog(blogInContext)
+            )
+        }, completion: { result in
+            switch result {
+            case let .success(value):
+                if value.oldValue == shared {
+                    success?()
+                    return
+                }
```

Comment:
Have you considered using `guard ` to ensure the compiler reminds us to early return?

 ```suggestion
                 guard value.oldValue != shared else {
                     success?()
                     return
                 }
```

Reply:
I find the if statement reads more naturally: "if the old value is the same as the updated value, then we should call the success block and exist early". Whereas the double negative in the guard statement feels less nature: "make sure the old value is not the same as the updated value, otherwise we should call the success block and exist early".